### DISCUSSION TITLE
Why do you return MPV_EVENT_NONE2 when mpv_event_id is MPV_EVENT_NONE in Swift

### DISCUSSION THREAD

**Discussion by internelp** [mpv-player/mpv#14899] [Q&A] [answered]
Hi,

When I was developing a macOS project using Xcode, I found a log entry with the ID `MPV_EVENT_NONE2`.

```
Optional(__C.mpv_event(event_id: MPV_EVENT_NONE2, error: 0, reply_userdata: 0, data: nil))
```

I did not find this ID in the `client.h` file in `enum mpv_event_id`.
But I found it in this file, which makes me a bit confused.

https://github.com/mpv-player/mpv/blob/deb9d30e905b7b4645f18a26e6274a2859221631/osdep/mac/swift_extensions.swift

```
line 59: case MPV_EVENT_NONE: return "MPV_EVENT_NONE2"
```

Is there any special consideration when returning this ID?

Thanks

**Comment by Akemi** [comment, marked as answer]
this is just a typo, on german keyboards `shift+2 = "`. probably hit it twice by accident. though it won't break anything. it's only internally used as an dict index, or rather adding a string representation to the `mpv_event_id` struct (if you 'cast' to string, print it).

though we don't really print that anywhere, so it must be from your end.

anyway in this case a raw `mpv_event` was printed with `print()`. the `event_id` struct value is of type `mpv_event_id`. since the string representation of `mpv_event_id` was implemented in that extension, `print()` is able to pretty print it. without that extension it would look like: `mpv_event(event_id: __C.mpv_event_id(rawValue: 0), error: 0, reply_userdata: 0, data: nil)`.

**Reply by internelp** [reply]
Thanks for your reply

**Reply by Akemi** [reply]
you are welcome.
